tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -491,295 +795,70 @@ async def create_selection(self, source, index=None, name=None):
-                            computation_types.FederatedType)
-    type_utils.check_equivalent_types(pre_report.type_signature.member,
-                                      report_type.parameter)
-
-    report = arg.internal_representation[4]
-    return await self._compute_intrinsic_federated_apply(
-        FederatingExecutorValue(
-            anonymous_tuple.AnonymousTuple([
-                (None, report), (None, pre_report.internal_representation)
-            ]),
-            computation_types.NamedTupleType(
-                (report_type, pre_report.type_signature))))
+    return await self.intrinsic_strategy.federated_aggregate(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_sum(self, arg):
-    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
-    zero, plus = tuple(await
-                       asyncio.gather(*[
-                           executor_utils.embed_tf_scalar_constant(
-                               self, arg.type_signature.member, 0),
-                           executor_utils.embed_tf_binary_operator(
-                               self, arg.type_signature.member, tf.add)
-                       ]))
-    return await self._compute_intrinsic_federated_reduce(
-        FederatingExecutorValue(
-            anonymous_tuple.AnonymousTuple([
-                (None, arg.internal_representation),
-                (None, zero.internal_representation),
-                (None, plus.internal_representation)
-            ]),
-            computation_types.NamedTupleType(
-                (arg.type_signature, zero.type_signature, plus.type_signature)))
-    )
+    return await self.intrinsic_strategy.federated_sum(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_mean(self, arg):
-    arg_sum = await self._compute_intrinsic_federated_sum(arg)
-    member_type = arg_sum.type_signature.member
-    count = float(len(arg.internal_representation))
-    if count < 1.0:
-      raise RuntimeError('Cannot compute a federated mean over an empty group.')
-    child = self._target_executors[placement_literals.SERVER][0]
-    factor, multiply = tuple(await asyncio.gather(*[
-        executor_utils.embed_tf_scalar_constant(child, member_type,
-                                                float(1.0 / count)),
-        executor_utils.embed_tf_binary_operator(child, member_type, tf.multiply)
-    ]))
-    multiply_arg = await child.create_tuple(
-        anonymous_tuple.AnonymousTuple([(None,
-                                         arg_sum.internal_representation[0]),
-                                        (None, factor)]))
-    result = await child.create_call(multiply, multiply_arg)
-    return FederatingExecutorValue([result], arg_sum.type_signature)
+    return await self.intrinsic_strategy.federated_mean(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_weighted_mean(self, arg):
     return await executor_utils.compute_federated_weighted_mean(self, arg)
```

Comment:
The implementation here should be delegated to the strategy as well. In the short term the strategy will still delegate to this function, but that can probably eventually be inlined.